plugins/lua/zone.lua

```diff
@@ -562,16 +575,21 @@ function AssignAnimal:cache_choices()
     for _, unit in ipairs(df.global.world.units.active) do
         if not is_assignable_unit(unit) then goto continue end
         local raw = df.creature_raw.find(unit.race)
+        local desc = dfhack.units.getReadableName(unit)
+        if type(unit.custom_profession) == 'string' and #unit.custom_profession > 0 then
+            desc = unit.custom_profession .. ', ' .. desc
```

Comment:
Another option is to attempt to split by `", "` in lua code, to insert custom profession after the name.

Although I agree this is little strange, that profession goes before the name, but that's the info I care about the most, since I type this manually. Game-generated names of pets are just random gibberish that doesn't help to pick animals for zones.

Reply:
That is a fair point. For the people who *do* use professions to categorize their livestock, they'll want it first so it can determine the sort order. Let me load this into my game and see what it looks like with prisoners and citizens and such